kong/globalpatches.lua

```diff
@@ -288,35 +288,36 @@ return function(options)
     local old_tcp_connect
     local old_udp_setpeername
 
-    local function tcp_resolve_connect(sock, host, port, sock_opts)
-      local target_ip, target_port = toip(host, port)
-      if not target_ip then
-        return nil, "[toip() name lookup failed]: " .. tostring(target_port) -- err
+  -- need to do the extra check here: https://github.com/openresty/lua-nginx-module/issues/860
+    local function strip_nils(port, opts)
+      if port and opts then
+        return port, opts
+      elseif opts then
+        return opts
+      elseif port then
+        return port
       end
-
-      -- need to do the extra check here: https://github.com/openresty/lua-nginx-module/issues/860
-      if not sock_opts then
-        return old_tcp_connect(sock, target_ip, target_port)
-      end
-
-      return old_tcp_connect(sock, target_ip, target_port, sock_opts)
     end
 
-    local function udp_resolve_setpeername(sock, host, port)
+    local function resolve_connect(f, sock, host, port, sock_opts)
       local target_ip, target_port
-
       if sub(host, 1, 5) == "unix:" then
-        target_ip = host -- unix domain socket, so just maintain the named values
-
+        target_ip = host
       else
         target_ip, target_port = toip(host, port)
-
-        if not target_ip then
-          return nil, "[toip() name lookup failed]: " .. tostring(target_port) -- err
-        end
       end
+      if not target_ip then
```

Comment:
this check can be moved in the `else` block above, since it doesn't apply to unix sockets